src/java/org/apache/cassandra/db/compaction/CompactionManager.java

```diff
@@ -1865,6 +1865,23 @@ public void setMaximumValidatorThreads(int number)
         validationExecutor.setMaximumPoolSize(number);
     }
 
+    public Map<String, Integer> getPendingTasksByKeyspaceAndTable()
+    {
+        Map<String, Integer> pendingCompactions = new HashMap<>();
+        for (String keyspaceName : Schema.instance.getKeyspaces())
+        {
+            for (ColumnFamilyStore cfs : Keyspace.open(keyspaceName).getColumnFamilyStores())
+            {
+                int estimatedRemainingTasks = cfs.getCompactionStrategy().getEstimatedRemainingTasks();
+                if (estimatedRemainingTasks > 0)
+                {
+                    pendingCompactions.put(String.format("%s/%s", keyspaceName, cfs.getColumnFamilyName()), estimatedRemainingTasks);
+                }
+            }
+        }
+        return pendingCompactions;
+    }
```

Comment:
This can be a stream but it's ugly due to the nested for, pref to keep it as is

```java
public Map<String, Integer> getPendingTasksByKeyspaceAndTable() {
    return Schema.instance.getKeyspaces().stream()
        .flatMap(keyspaceName -> Keyspace.open(keyspaceName).getColumnFamilyStores().stream()
            .map(cfs -> new AbstractMap.SimpleEntry<>(
                String.format("%s/%s", keyspaceName, cfs.getColumnFamilyName()),
                cfs.getCompactionStrategy().getEstimatedRemainingTasks()
            ))
        )
        .filter(entry -> entry.getValue() > 0)
        .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue));
}
```

Reply:
Not opposed to keeping the for loop